src/core/modules/Nebula.cpp

```diff
@@ -811,13 +808,68 @@ void Nebula::drawHints(StelPainter& sPainter, float maxMagHints, StelCore *core)
 		case NebEn:
 		case NebSNC:
 		case NebSNRC:
-			if (finalSize > 75.f)
-				Nebula::texDiffuseNebulaXLarge->bind();
-			else if (finalSize > 35.f)
-				Nebula::texDiffuseNebulaLarge->bind();
-			else
-				Nebula::texDiffuseNebula->bind();
-			break;
+		{
+			// Take into account device pixel density and global scale ratio, as we are drawing 2D stuff.
+			const auto pixelRatio = sPainter.getProjector()->getDevicePixelsPerPixel();
+			const auto scale = pixelRatio * StelApp::getInstance().getGlobalScalingRatio();
+			finalSize *= scale;
+			// FIXME: this size reduction is just for compatibility with the
```

Comment:
I guess the 0.89 ensures that the box line is as large as the blurred line that had to be technically inside the icon's pixels.  We can of course get rid of it. The symbol should approximately represent the size given in the data. (But keep a minimum size for small objects or large fov.)